src/compiler/moduleSpecifiers.ts

```diff
@@ -594,7 +594,9 @@ function getLocalModuleSpecifier(moduleFileName: string, info: Info, compilerOpt
         return pathsOnly ? undefined : relativePath;
     }
 
-    const baseDirectory = getNormalizedAbsolutePath(getPathsBasePath(compilerOptions, host) || baseUrl!, host.getCurrentDirectory());
+    const currentDirectory = host.getCurrentDirectory();
+    const basePath = getPathsBasePath(compilerOptions, host) ?? baseUrl ?? currentDirectory;

```

Comment:
We incorrectly assumed `baseUrl` would be defined; however, the user could incorrectly have set `paths` and not `baseUrl`.